frontend/src/components/common/Notifications/index.tsx

```diff
@@ -0,0 +1,172 @@
+          </Grid>
+          <Grid item>{timeAgo(notification.date)}</Grid>
+        </Grid>
+      </MenuItem>
+    );
+  });
+}
+
+export default function Notifications() {
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const history = useHistory();
+  const notifications = useTypedSelector(state => state.ui.notifications);
+  const [notificationAlert, setNotificationAlert] = useState(false);
+  const dispatch = useDispatch();
+  const [events, error] = Event.useList();
+
+  useEffect(() => {
+    if (events && events.length !== 0) {
+      const importantEvents: Notification[] = events
+        .filter((event: KubeEvent) => event.type !== 'Normal')
+        .sort(
+          (e1: KubeEvent, e2: KubeEvent) =>
+            new Date(e2.metadata.creationTimestamp).getTime() -
+            new Date(e1.metadata.creationTimestamp).getTime()
+        )
+        .map((event: KubeEvent) => {
+          const notification = new Notification();
+          notification.date = event.metadata.creationTimestamp;
+          notification.id = event.metadata.uid;
+          notification.message = event.message;
+          notification.url = '/';
+          return notification;
+        });
+      // we are here means the events list changed and we have now new set of events, so we will notify the store about it
+      dispatch(setUINotifications(importantEvents));
+    }
+  }, [events, error]);
+
+  useEffect(() => {
+    if (!notifications || notifications.length === 0) {
+      return;
+    }
+    // push notification only if we have some cluster selected + there are events
+    const recentNotificationFromStorage = localStorage.getItem('recent_notification');
+    if (recentNotificationFromStorage) {
+      const parsedRecentNotification = JSON.parse(recentNotificationFromStorage);
+      if (parsedRecentNotification?.id !== notifications[0].id) {
+        setNotificationAlert(true);
+      }
+    }
+    // we store the recent event in storage which we will use to check if there were new notifications loaded from last time the user checked the notification tab
+    localStorage.setItem('recent_notification', JSON.stringify(notifications[0]));
+  }, [notifications]);
+
+  const handleClick = (event: any) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+    setNotificationAlert(false);
+  };
+
+  function menuItemClickHandler(url?: string) {
+    setAnchorEl(null);
+    setNotificationAlert(false);
+    //only take us to events page if it's not a error notification from fetching events
+    if (url) {
+      history.push(url);
+    }
+  }
+
+  return (
+    <>
+      <IconButton onClick={handleClick}>
+        {notificationAlert ? (
+          <Badge variant="dot" color="error">
+            <Icon icon={bellIcon} />
+          </Badge>
+        ) : (
+          <Icon icon={bellIcon} />
+        )}
+      </IconButton>
+      <Menu
+        id="simple-menu"
```

Comment:
Is this id used for anything? I know this component should probably only be used once at the moment... (so it's not that important). But if this component is used multiple times this would be a problem repeating the same id.